java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -1762,6 +1762,37 @@ public static ColumnVector stringConcatenate(Scalar separator, Scalar narep, Col
     }
   }
 
+  /**
+   * Interleave an array of columns into a single column.
+   *
+   * Example:
+   * ```
+   * input  = [[A1, A2, A3], [B1, B2, B3]]
+   * return = [A1, B1, A2, B2, A3, B3]
+   * ```
+   *
+   * @param columns Columns to interleave.
+   *
+   * @return The interleaved columns as a single column
+   */
+  public static ColumnVector interleaveColumns(ColumnVector[] columns) {
```

Comment:
Should this be a `Table` API rather than a `ColumnVector` API?  The underlying libcudf API wants a table anyway, so that would be a more natural mapping of the binding.

Reply:
Note that it would also help make it apparent to the end-user that all the columns have to be the same length.

Reply:
Sounds good. Is there an existing example of this pattern? I'm having trouble finding something like that.

Reply:
Not sure what you mean by "this pattern."  Do you mean `Table` APIs that can return a `ColumnVector`?  If so, see `Table`'s `lowerBound` method for one example.

Reply:
I'm not sold on the table API. I personally find it more cumbersome to do 
```
try (Table tmp = new Table(columns)) {
  tmp.interleaveColumns();
}
```

compared to 

```
ColumnVector.interleaveColumns(columns);
```

But I am okay with it because it does reflect the underlying cudf API.

Reply:
I have now converted this to the table API.